std/meta.d

```diff
@@ -1309,6 +1309,65 @@ private template staticMerge(alias cmp, int half, Seq...)
     }
 }
 
+/**
+ * Allows subsetting of alias sequence that are not
+ * necessarily adjacent to each other, i.e. cannot be sliced.
+ *
+ * Parameters:
+ *  range = a range object preferably an unsigned array to index T
+ *  T = a set of types to be indexed
+ *
+ * Returns: alias sequence of selected types
+ *
+ * Evaluates to `AliasSeq!(T[range[0]], T[range[1]], ..., T[range[n]])`.
+ *
+ * How:
+ *      alias T1 = staticSubset!([0, 2, 1, 1, 2, 0], int, char, double);
+ */
+template staticSubset(alias range, T...)
+{
+    import std.traits : isArray, isNarrowString;
+
+    alias ArrT = typeof(range);
+    static if (isArray!ArrT && !isNarrowString!ArrT)
+    {
+        static if (range.length == 0)
+        {
+            alias staticSubset = AliasSeq!();
+        }
+        else
+        {
+            alias staticSubset = AliasSeq!(T[range[0]], staticSubset!(range[1 .. $], T));
+        }
+    }
+    else
+    {
+        import std.range.primitives : isInputRange;
+        static if (isInputRange!ArrT)
+        {
+            import std.array : array;
+            alias staticSubset = staticSubset!(array(range));
```

Comment:
Don't you need the `T` template parameter here? There's no unit test here for the range version.